### PR TITLE
Add resource response functionality

### PR DESCRIPTION
**Pull request by mondragonfx**
## Description

PR for adding return response for vultr resources to govultr.

### Checklist:

* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Have you linted your code locally prior to submission?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by happytreees — APPROVED** [review]
This PR introduces breaking changes to the package for any endusers

File: backup.go [hunk 1]
```diff
@@ -72,7 +73,7 @@ func (b *BackupServiceHandler) List(ctx context.Context, options *ListOptions) (
 	req.URL.RawQuery = newValues.Encode()
 
 	backups := new(backupsBase)
-	if err = b.client.DoWithContext(ctx, req, backups); err != nil {
+	if _, err = b.client.DoWithContext(ctx, req, backups); err != nil {
```
Comment (christhemorse) [inline backup.go:76]: Shouldn't this function be returning the API json response as well? Backup -> List()

Comment (mondragonfx) [reply backup.go]: @christhemorse That is correct. all set.

**Review by happytreees — APPROVED** [review]
lgtm

**Review by christhemorse — APPROVED** [review]
Looks good after recent changes. Nothing else has jumped out at me and the approach for passing along the API json response should help enable a json feature for vultr-cli and other utilities, and give folks using the library more flexibility to use that data if they need it. Approved.

**Pull request merged**